rust/nix-expr/src/eval_state.rs

```diff
@@ -191,12 +237,24 @@ impl Drop for EvalState {
 #[cfg(test)]
 mod tests {
     use ctor::ctor;
+    use nix_util::settings;
 
     use super::*;
 
     #[ctor]
     fn setup() {
-        init().unwrap();
+        (|| -> Result<()> {
+            init()?;
+            // If it reinvokes the test suite,
+            // settings::set("build-hook", "")?;
+
+            // When testing in the sandbox, the default build dir would be a parent of the storeDir,
+            // which causes an error. So we set a custom build dir here.
+            settings::set("sandbox-build-dir", "/custom-build-dir-for-test")?;
```

Comment:
This is using our bindings in order to change settings while nix is running on this repo, right?

I'm not even sure how that works, since we're not the ones driving the nix build in this case.

In any case, does it not make more sense to put that in the `.nix` files that are driving the build?

Reply:
It only applies during testing.

I'd like to keep the `.nix` files simple, so that it's easier to re-package for Nixpkgs, or as part of a different Rust-to-Nix integration.

Reply:
Aha! As we discussed, this is not operating on the system nix, but rather on the nix that we're linking in. No problem in that case.